pkg/customresource/customresource.go

```diff
@@ -157,10 +157,13 @@ func rawCRDFromFile(path string) ([]byte, error) {
 	return yamls.ReadFile(path)
 }
 
-func waitForCRDInit(context Context, resource CustomResource) error {
+func waitForCRDInit(crcontext Context, resource CustomResource) error {
 	crdName := fmt.Sprintf("%s.%s", resource.Plural, resource.Group)
-	return wait.Poll(context.Interval, context.Timeout, func() (bool, error) {
-		crd, err := context.APIExtensionClientset.ApiextensionsV1().CustomResourceDefinitions().Get(contextpkg.TODO(), crdName, metav1.GetOptions{})
+	deadlineCtx, cancel := contextpkg.WithTimeout(contextpkg.TODO(), crcontext.Timeout)
+	defer cancel()
+
+	return wait.PollUntilContextTimeout(deadlineCtx, crcontext.Interval, crcontext.Timeout, false, func(ctx contextpkg.Context) (bool, error) {
```

Comment:
I don't think we need to create context with timeout `deadlineCtx`. The arg `crcontext.Timeout` keep track of timeout. Ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextTimeout

We are not creating context WithTimeout at other places e.g https://github.com/kanisterio/kanister/pull/2532/files#diff-978a334f27b9a6b8238cf2091c750207e74b46a228b26707d0d03a0ab08d00e9R339

Reply:
Done